src/App.tsx

```diff
@@ -1,29 +1,64 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.css';
 import { getNumbers } from './utils';
+import cn from 'classnames';
 
 // eslint-disable-next-line @typescript-eslint/no-unused-vars
 const items = getNumbers(1, 42).map(n => `Item ${n}`);
 
+type Option = '3' | '5' | '10' | '20';
+
+const options: Option[] = ['3', '5', '10', '20'];
+
 export const App: React.FC = () => {
+  const [perPage, setPerPage] = useState<Option>(options[1]);
+  const [currentPage, setCurrentPage] = useState<number>(1);
+  const amountOfPages = Math.ceil(items.length / Number(perPage));
+
+  const firstIndex = (currentPage - 1) * Number(perPage);
+  const lastIndex = firstIndex + Number(perPage);
+  const visibleItems = items.slice(firstIndex, lastIndex);
+  const lastItemIndex = Math.min(lastIndex, items.length);
+
+  function handleNextPage(index: number) {
+    if (index + 1 <= amountOfPages) {
+      setCurrentPage(current => current + 1);
+    }
+  }
+
+  function handlePreviousPage(index: number) {
+    if (index - 1 >= 1) {
+      setCurrentPage(current => current - 1);
+    }
+  }
+
+  function handleItemsPerPage(value: Option) {
+    setPerPage(value as Option);
+    setCurrentPage(1);
+  }
+
   return (
     <div className="container">
       <h1>Items with Pagination</h1>
 
       <p className="lead" data-cy="info">
-        Page 1 (items 1 - 5 of 42)
+        {`Page ${currentPage} (items ${firstIndex + 1} - ${lastItemIndex} of ${items.length})`}
       </p>
 
       <div className="form-group row">
         <div className="col-3 col-sm-2 col-xl-1">
           <select
+            defaultValue="5"
```

Comment:
As i know, it's better not to use `defaultValue` when you're also managing its state. It'd be better to use `value` with element from state.

Something like this:
```suggestion
            value={perPage}
```